### DISCUSSION TITLE
Excessive swap usage

### DISCUSSION THREAD

**Discussion by erincerys** [phuhl/linux_notification_center#249] [Q&A]
Anyone got an idea as to what I can do about this bugger using 3.5GB swap sometimes?

I never "delete" notifications. They just pile up. It takes like 5 seconds for the panel to merely load and open when I hit the keybind, in this "overloaded" state.

Surely there's a way to impose a limit, yes? Automatically discarding old messages, and not loading more than a certain number in the panel...

**Comment by phuhl** [comment]
Jeeze, how many notifications do you have open at the same time?

**Reply by erincerys** [reply]
Is there a way to get a count without doing it manually?

Probably thousands per day between DMs, email, and download activity.

**Reply by phuhl** [reply]
I am afraid, currently not.

I guess, to support that many notifications, we'd have to build a virtual list into the notification center, to not render all notifications, but only the ones that fit on screen. But that is an effort, that I cannot go into. PRs are welcome, though.
